tests/filecheck/dialects/riscv/riscv_assembly_emission.mlir

```diff
@@ -1,178 +1,187 @@
-  // CHECK-NEXT: csrrw j1, 1024, j2
-  %csrrw_w = "riscv.csrrw"(%2) {"csr" = 1024 : i32, "writeonly"}: (!riscv.reg<j2>) -> !riscv.reg<zero>
-  // CHECK-NEXT: csrrw zero, 1024, j2
-  %csrrs_rw = "riscv.csrrs"(%2) {"csr" = 1024 : i32}: (!riscv.reg<j2>) -> !riscv.reg<zero>
-  // CHECK-NEXT: csrrs zero, 1024, j2
-  %csrrs_r = "riscv.csrrs"(%0) {"csr" = 1024 : i32, "readonly"}: (!riscv.reg<zero>) -> !riscv.reg<j2>
-  // CHECK-NEXT: csrrs j2, 1024, zero
-  %csrrc_rw = "riscv.csrrc"(%2) {"csr" = 1024 : i32}: (!riscv.reg<j2>) -> !riscv.reg<j0>
-  // CHECK-NEXT: csrrc j0, 1024, j2
-  %csrrc_r = "riscv.csrrc"(%0) {"csr" = 1024 : i32, "readonly"}: (!riscv.reg<zero>) -> !riscv.reg<j0>
-  // CHECK-NEXT: csrrc j0, 1024, zero
-  %csrrsi_rw = "riscv.csrrsi"() {"csr" = 1024 : i32, "immediate" = 8 : i32}: () -> !riscv.reg<j1>
-  // CHECK-NEXT: csrrsi j1, 1024, 8
-  %csrrsi_r = "riscv.csrrsi"() {"csr" = 1024 : i32, "immediate" = 0 : i32}: () -> !riscv.reg<j0>
-  // CHECK-NEXT: csrrsi j0, 1024, 0
-  %csrrci_rw = "riscv.csrrci"() {"csr" = 1024 : i32, "immediate" = 8 : i32}: () -> !riscv.reg<j0>
-  // CHECK-NEXT: csrrci j0, 1024, 8
-  %csrrci_r = "riscv.csrrci"() {"csr" = 1024 : i32, "immediate" = 0 : i32}: () -> !riscv.reg<j1>
-  // CHECK-NEXT: csrrci j1, 1024, 0
-  %csrrwi_rw = "riscv.csrrwi"() {"csr" = 1024 : i32}: () -> !riscv.reg<j0>
-  // CHECK-NEXT: csrrwi j0, 1024
-  %csrrwi_w = "riscv.csrrwi"() {"csr" = 1024 : i32, "writeonly"}: () -> !riscv.reg<zero>
-  // CHECK-NEXT: csrrwi zero, 1024
-
-  // Assembler pseudo-instructions
-  %li = "riscv.li"() {"immediate" = 1 : i32}: () -> !riscv.reg<j0>
-  // CHECK-NEXT: li j0, 1
-  // Environment Call and Breakpoints
-  "riscv.ecall"() : () -> ()
-  // CHECK-NEXT: ecall
-  "riscv.ebreak"() : () -> ()
-  // CHECK-NEXT: ebreak
-  "riscv.directive"() {"directive" = ".align", "value" = "2"} : () -> ()
-  // CHECK-NEXT: .align 2
-  "riscv.directive"() ({
-    %nested_addi = "riscv.addi"(%1) {"immediate" = 1 : i32}: (!riscv.reg<j1>) -> !riscv.reg<j1>
-  }) {"directive" = ".text"} : () -> ()
-  // CHECK-NEXT:  .text
-  // CHECK-NEXT:  addi j1, j1, 1
-  "riscv.label"() {"label" = #riscv.label<"label0">} : () -> ()
-  // CHECK-NEXT: label0:
   "riscv.label"() ({
-    %nested_addi = "riscv.addi"(%1) {"immediate" = 1 : i32}: (!riscv.reg<j1>) -> !riscv.reg<j1>
-  }) {"label" = #riscv.label<"label1">} : () -> ()
-  // CHECK-NEXT: label1:
-  // CHECK-NEXT: addi j1, j1, 1
-
-
-  // Custom instruction
-  %custom0, %custom1 = "riscv.custom_assembly_instruction"(%0, %1) {"instruction_name" = "hello"} : (!riscv.reg<zero>, !riscv.reg<j1>) -> (!riscv.reg<j3>, !riscv.reg<j4>)
-  // CHECK-NEXT:   hello j3, j4, zero, j1
-
-  // RV32I/RV64I: 2.5 Control Transfer Instructions (cont'd)
-  // terminators
-
-  // Unconditional Branch Instructions
-  "riscv.ret"() : () -> () // pseudo-instruction
-  // CHECK-NEXT: ret
+    %0 = "riscv.li"() {"immediate" = 6 : i32} : () -> !riscv.reg<zero>
```

Comment:
The first block is not printed if it doesn't have arguments, and only block that are printed get labels. It's sort of annoying, I'd like to fix it in the future, but right now that's how it is. I took the labels that xdsl-opt printed and put them on the input so that it wouldn't be confusing.